python/tutorials/gluon/08-warp-specialization.py

```diff
@@ -0,0 +1,655 @@
+"""
+Warp Specialization
+===================
+
+This tutorial covers warp specialization. In typical GPU kernels, all warps are
+executing cooperatively, meaning they perform parts of the same task. Warp
+specialization, however, is a technique where different warps in the kernel are
+doing completely different tasks.
```

Comment:
Does "cooperatively" imply performing the exact same task? e.g. I would say the warps in a warp-specialized kernel are cooperating.

Reply:
My understanding of kernel programming (NVIDIA) terminology is that "cooperative" implies they are doing the same work. I can tweak the terminology here to be more explicit.